platform/pulpcore/app/serializers/base.py

```diff
@@ -144,6 +143,39 @@ def _populate_generic_fields(self, instance, field_mappings):
             field.mapping.replace(mapping)
 
 
+class NestedModelSerializer(ModelSerializer):
+    """
+    This class allows the serializer to have WritableNestedUrlRelated fields, which represent
+    nested parent objects that are implied by the url of a request.
+    """
+
+    def to_internal_value(self, *args, **kwargs):
+        """
+        After letting DRF populate and validate all writable fields, we populate and validate
+        `href_writable` fields.
+        """
+        # super().to_internal_value will validate all writable fields and populate serializer.data
+        validated_data = super().to_internal_value(*args, **kwargs)
+        # nested parent fields should be read_only (so they don't accept args that are ignored)
+        # instead, validate the parent based on the parent url and add it to the validated data.
+        validated_data.update(self.writable_nested_parent_url_to_internal())
+        return validated_data
+
+    def writable_nested_parent_url_to_internal(self):
+        """
+        Returns an ordered dict containing each `href_writable` field and its internal value.
+        """
+        url_parents = OrderedDict()
+        for field in self.fields.values():
+            # Writable nested parent objects must be read only (parent is regrieved using kwargs)
+            if (getattr(field, 'href_writable', False)):
+                # Use the original viewset to retrieve the parent object implied by the request url
+                # TODO(asmacdo) catch the 404
+                parent_field, parent_obj = self.context['view'].get_parent_field_and_object()
```

Comment:
Everything is an object in python, what is the the `_obj` suffix trying to communicate?